File: frontendapp/lib/models/room.dart
class Room {
  final int? id;
  final int cinemaId;
  final String name;
  final String seatRows;
  final int seatRowMax;

  Room({
    this.id,
    required this.cinemaId,
    required this.name,
    required this.seatRows,
    required this.seatRowMax,
  });

  factory Room.fromMap(Map<String, dynamic> map) => Room(
    id: map['ID'],
    cinemaId: map['CINEMAID'],
    name: map['NAME'],
    seatRows: map['SEATROWS'],
    seatRowMax: map['SEATROWMAX'],
  );

  Map<String, dynamic> toMap() => {
    'ID': id,
    'CINEMAID': cinemaId,
    'NAME': name,
    'SEATROWS': seatRows,
    'SEATROWMAX': seatRowMax,
  };
}

File: backend/bookingmovieticketapp/src/main/java/com/project/bookingmovieticketapp/Models/Room.java
package com.project.bookingmovieticketapp.Models;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "rooms")
@Builder
public class Room {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name = "cinemaid")
    private Cinema cinema;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "seatcolumnmax",nullable = false)
    private int seatcolumnmax;

    @Column(name="seatrowmax",nullable = false)
    private int seatrowmax;
}
